### PR TITLE
feat: update external reader interface to use a single replica

### PR DESCRIPTION
**Pull request by cryptonemo**

### CODE REVIEW

File: src/merkle.rs [hunk 1]
```diff
@@ -201,25 +201,26 @@ impl<
     MerkleTree<E, A, LevelCacheStore<E, std::fs::File>, BaseTreeArity, SubTreeArity, TopTreeArity>
 {
     /// Given a pathbuf, instantiate an ExternalReader and set it for the LevelCacheStore.
-    pub fn set_external_reader_path(&mut self, path: &PathBuf) -> Result<()> {
+    pub fn set_external_reader_path(&mut self, path: &PathBuf, offset: usize) -> Result<()> {
```
Comment (dignifiedquire) [inline src/merkle.rs:204]: could we keep supporting without offset, to avoid breaking the API?

File: src/store/mod.rs [hunk 2]
```diff
@@ -91,6 +93,21 @@ pub enum StoreConfigDataVersion {
 
 const DEFAULT_STORE_CONFIG_DATA_VERSION: u32 = StoreConfigDataVersion::Two as u32;
 
+#[derive(Clone, Debug, Serialize, Deserialize, Default)]
+pub struct ReplicaConfig {
```
Comment (dignifiedquire) [inline src/store/mod.rs:97]: you could implement `From<PathBuf> for ReplicaConfig` setting a single offset at `0` to keep easily supporting the existing usage.

File: src/merkle.rs [hunk 3]
```diff
@@ -258,15 +256,16 @@ impl<
...
+    /// ordering of the trees is significant, as trees are leaf
+    /// indexed / addressable in the same sequence that they are
+    /// provided here.
     #[allow(clippy::type_complexity)]
-    pub fn from_sub_tree_store_configs_and_replicas(
+    pub fn from_sub_tree_store_configs_and_replica(
         leafs: usize,
         configs: &[StoreConfig],
-        replica_paths: &[PathBuf],
+        replica_config: &ReplicaConfig,
```
Comment (dignifiedquire) [inline src/merkle.rs:268]: Could we support taking `&[ReplicaConfig]`, allowing for different paths,

**Comment by cryptonemo** [comment]
@dignifiedquire Merge and version bump please

**Pull request merged**